### PR TITLE
[WFCORE-4135] Support for read only server configuration directory

### PR DESCRIPTION
**Pull request by yersan**
A proposal to use read only server configuration dir

Jira issue: https://issues.redhat.com/browse/WFCORE-4135

### CODE REVIEW

File: ee/WFCORE-4135_Support_for_ read_only_server_config_dir.adoc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+The goal of this feature is to make `--read-only-server-config=<server config file name>` setting work in standalone mode even if the whole configuration directory pointed by `jboss.server.config` is a read only directory.
+
+Domain mode changes are not initially included on this feature.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-4135[WFCORE-4135]
```
Comment (jmesnil) [inline ee/WFCORE-4135_Support_for_ read_only_server_config_dir.adoc:33]: The proposal is missing a link to the EAP7 RFE https://issues.redhat.com/browse/EAP7-1119

Comment (yersan) [reply ee/WFCORE-4135_Support_for_ read_only_server_config_dir.adoc:34]: Thanks, I've just added the link.

**Comment by yersan** [comment]
Hello @jmesnil , when you have a chance, could you review again and approve in case? Test plan has been merged to Polarion

**Comment by yersan** [comment]
Sorry @jmesnil , hold on the review a bit more, I still need to see how we are going to adapt the cloud image launch to use the `--read-only-server-config=<server config file name>` in terms of Jira handling and possible code changes, I'm going to check it today.

**Comment by yersan** [comment]
Hello @jmesnil @OndrejKotek, 

As discussed with you, I just updated the proposal to describe a bit the cloud images changes. 

Initially I thought we should enable the feature by using an environment variable in the cloud images, but later I thought it would be better if the image detects it and it's auto-configured by itself. Secondly, pay attention to the markers availability, It sounds reasonable to me to not allow  them under this scenario.

 Please could you take a look and share feedback/approval?

Thanks!

File: ee/WFCORE-4135_Support_for_ read_only_server_config_dir.adoc [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+The goal of this feature is to make `--read-only-server-config=<server config file name>` setting work in standalone mode even if the whole configuration directory pointed by `jboss.server.config` is a read only directory.
+
+Domain mode changes are not initially included on this feature.
+
+== Required changes for cloud images
+
+The cloud images should be smart enough to detect if the server configuration directory is a read only directory, and in this case, the launch command for the underlying server should incorporate the `--read-only-server-config=<server config file name>` argument to make it possible to start the server under this circumstance.
```
Comment (OndrejKotek) [inline ee/WFCORE-4135_Support_for_ read_only_server_config_dir.adoc:31]: This should be captured in _Hard Requirements_

Comment (yersan) [reply ee/WFCORE-4135_Support_for_ read_only_server_config_dir.adoc]: Thanks for reviewing @OndrejKotek, I've just added them as hard requirements.

**Comment by yersan** [comment]
I've added `IS_READ_ONLY_SERVER_CONFIG` to allow users control the behaviour manually if they want to append the `--read-only-server-config=<server config file name>`

File: ee/WFCORE-4135_Support_for_ read_only_server_config_dir.adoc [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+WildFly Core
+WildFly Cekit Modules
+
+== Requirements
+
+=== Hard Requirements
+
+* The `--read-only-server-config=<server config file name>` setting will be expanded to allow, not only a read only configuration file, but also a read only configuration directory.
+* The user can activate/deactivate this setting by using the following environment variable `IS_READ_ONLY_SERVER_CONFIG=<true|false>`
```
Comment (OndrejKotek) [inline ee/WFCORE-4135_Support_for_ read_only_server_config_dir.adoc:72]: I expect the last three points are just for OCP environments.

Comment (yersan) [reply ee/WFCORE-4135_Support_for_ read_only_server_config_dir.adoc]: Yes, you're right. Done.

**Comment by OndrejKotek** [comment]
@yersan Can you update the test plan section and the RFE test plan please? I have no more suggestions to the rest of the content.

**Comment by yersan** [comment]
Closing by now, I have no plans to work on this soon

**Comment by yersan** [comment]
superseded by https://github.com/wildfly/wildfly-proposals/pull/342

**Pull request closed without merging**